src/Symfony/Component/Serializer/Normalizer/ConstraintViolationListNormalizer.php

```diff
@@ -57,6 +63,17 @@ public function normalize($object, string $format = null, array $context = [])
                 $violationEntry['type'] = sprintf('urn:uuid:%s', $code);
             }
 
+            $constraint = $violation->getConstraint();
+            if (
+                null !== $payloadFieldsToSerialize &&
+                $constraint &&
+                $constraint->payload &&
+                // If some or all payload fields are whitelisted, add them
+                $payloadFields = [] === $payloadFieldsToSerialize || true === $payloadFieldsToSerialize ? $constraint->payload : array_intersect_key($constraint->payload, $payloadFieldsToSerialize)
```

Comment:
`!$payloadFieldsToSerialize`

Reply:
Hum, not, because $payloadFieldsToSerialize can be array|true.
But why make the empty array a special case? It contradicts with the array intersection just after.

Reply:
I'll change that to make `null` a special case as suggested by @teohhanhui in https://github.com/api-platform/core/pull/3375